src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -43,51 +42,212 @@ public ComplaintResponseService(ComplaintRepository complaintRepository, Complai
+        log.debug("Removed empty complaint and thus lock for complaint with id : {}", complaintId);
+    }
 
-        // Do not trust user input
-        Optional<Complaint> originalComplaintOptional = complaintRepository.findByIdWithEagerAssessor(complaintId);
-        if (originalComplaintOptional.isEmpty()) {
-            throw new BadRequestAlertException("The complaint you are referring to does not exist", ENTITY_NAME, "noresult");
+    private ComplaintResponse getUnfinishedComplaintResponse(Complaint complaintFromDatabase) {
+        if (complaintFromDatabase.isAccepted() != null) {
+            throw new IllegalArgumentException("Complaint is already handled");
         }
+        if (complaintFromDatabase.getComplaintResponse() == null) {
+            throw new IllegalArgumentException("Complaint response does not exists for given complaint");
+        }
+        if (complaintFromDatabase.getComplaintResponse().getSubmittedTime() != null) {
+            throw new IllegalArgumentException("Complaint response is already submitted");
+        }
+        ComplaintResponse unfinishedComplaintResponse = complaintFromDatabase.getComplaintResponse();
+        return unfinishedComplaintResponse;
+    }
 
-        Complaint originalComplaint = originalComplaintOptional.get();
+    /**
+     * Refreshes the empty complaint response for a given complaint that acts a lock
+     *
+     * The empty complaint response acts as a lock. Only the reviewer of the empty complaint response and instructors can resolve the complaint as long as the lock
+     * is running. For lock duration calculation see: {@link ComplaintResponse#isCurrentlyLocked()}. This methods exchanges the current empty complaint response to a new one
+     * thus updating the lock.
+     *
+     * This is possible in two cases:
+     *
+     * Case A: Lock is currently active -> Only the initial reviewer or an instructor can refresh the empty complaint response
+     * Case B: Lock has run out --> Others teaching assistants can refresh the empty complaint response thus acquiring the lock
+     *
+     * @param complaintId if of the complaint for which to refresh the empty response for
+     * @return refreshed empty complaint response
+     */
+    @Transactional // ok because of modifying query
+    public ComplaintResponse refreshEmptyComplaintResponse(Long complaintId) {
+        if (complaintId == null) {
+            throw new IllegalArgumentException("Complaint id should not be null");
+        }
+        Optional<Complaint> complaintFromDatabaseOptional = complaintRepository.findByIdWithEagerAssessor(complaintId);
+        if (complaintFromDatabaseOptional.isEmpty()) {
+            throw new IllegalArgumentException("Complaint was not found in database");
+        }
+        Complaint complaintFromDatabase = complaintFromDatabaseOptional.get();
+        ComplaintResponse unfinishedComplaintResponse = getUnfinishedComplaintResponse(complaintFromDatabase);
```

Comment:
Fwiw, this variable is unused.

Reply:
Removed it in d1fad87411ae6cb697c5f8ebafb872b41437ddc5